blivet/devices/partition.py

```diff
@@ -660,6 +660,20 @@ def resize(self):
         self.disk.format.commit()
         self.update_size()
 
+    @property
+    def protected(self):
+        protected = super().protected
+
+        # extended partition is protected also when one of its logical partitions is protected
+        if self.is_extended:
+            return any(part.protected for part in self.disk.children if part.is_logical) or protected
```

Comment:
``protected`` should be checked first here because if it's ``True`` we would avoid the more expensive check of all disk partitions.